library/system.lisp

```diff
@@ -74,57 +61,86 @@ While the result will always contain microseconds, some implementations may retu
 
 (coalton-toplevel
 
-  (declare get-run-time (Unit -> UFix))
+  (declare get-run-time (Unit -> Integer))
   (define (get-run-time)
-    "Gets the run-time."
-    (lisp UFix ()
+    "Gets the run-time in `internal time units`. This is implementation specific: it may measure real time, run time, CPU cycles, or some other quantity. The difference between two successive calls to this function represents quantity accumulated during that period of time."
+    (lisp Integer ()
       (cl:get-internal-run-time)))
 
-  (declare get-real-time (Unit -> UFix))
+  (declare get-real-time (Unit -> Integer))
   (define (get-real-time)
-    "Gets the real-time."
-    (lisp UFix ()
+    "Gets the real-time in `internal time units`. The difference between two successive calls to this function represents the time that has elapsed."
+    (lisp Integer ()
       (cl:get-internal-real-time)))
 
-  #+sbcl
-  (declare get-bytes-consed (Unit -> UFix))
-  #+sbcl
-  (define (get-bytes-consed)
-    "Gets the number of bytes consed (only implemented for SBCL"
-    (lisp UFix ()
-      (sb-ext:get-bytes-consed)))
+  (declare internal-time-units-per-second (Unit -> Integer))
+  (define (internal-time-units-per-second)
+    "The number of `internal time units` per second. This is implementation specific."
+    (lisp Integer ()
+      cl:internal-time-units-per-second))
+
+  (declare time-in-seconds (Integer -> Double-Float))
+  (define (time-in-seconds t)
+    "Converts `internal time units` into seconds."
+    (real:inexact/ t (internal-time-units-per-second)))
+
+  (declare time-in-microseconds (Integer -> Integer))
+  (define (time-in-microseconds t)
+    "Converts `internal time units` into microseconds."
+    (real:round/ (* 1000000 t)
+                 (internal-time-units-per-second)))
+
+  (declare monotonic-bytes-consed (Unit -> (Optional Integer)))
+  (define (monotonic-bytes-consed)
+    "Returns the number of bytes consed since some unspecified point in time.
+The difference between two successive calls to this function represents the number of bytes consed in that period of time."
+    #+sbcl
+    (Some (lisp Integer ()
+            (sb-ext:get-bytes-consed)))
+    #-sbcl
+    None)
 
   (define-struct (Profile :a)
-    "A profile of a run function."
+    "A profile of a function, containing space and timing information.."
     (output
-     "The output of the function" :a)
+     "The output of the function." :a)
     (run-time
```

Comment:
Maybe just store time units, and write a couple accessors for getting seconds etc.

It's strange to me to store inexact information in a profile capture.